ot/mpz-ot-core/src/ideal/ideal_spcot.rs

```diff
@@ -0,0 +1,116 @@
+//! Define ideal functionality of SPCOT.
+
+use mpz_core::{prg::Prg, Block};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+/// The message that sender receivers from the SPCOT functionality.
+pub struct SpcotMsgForSender {
+    /// The random blocks that sender receives from the SPCOT functionality.
+    pub v: Vec<Vec<Block>>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+/// The message that receiver receives from the SPCOT functionality.
+pub struct SpcotMsgForReceiver {
+    /// The random blocks that receiver receives from the SPCOT functionality.
+    pub w: Vec<Vec<Block>>,
+}
+
+#[allow(missing_docs)]
+pub struct IdealSpcot {
+    pub delta: Block,
+    pub counter: usize,
+    pub prg: Prg,
+}
+
+impl IdealSpcot {
+    /// Initiate the functionality.
+    pub fn init() -> Self {
```

Comment:
This is typically named `new`

```suggestion
    pub fn new() -> Self {
```